src/library_tty.js

```diff
@@ -153,8 +163,13 @@ mergeInto(LibraryManager.library, {
         if (val === null || val === {{{ charCode('\n') }}}) {
           Module['print'](UTF8ArrayToString(tty.output, 0));
           tty.output = [];
-        } else {
-          if (val != 0) tty.output.push(val); // val == 0 would cut text output off in the middle.
+        } else if (val != 0) {
+          tty.output.push(val); // val == 0 would cut text output off in the middle.
+        }
+      },
+      write: function(offset, length, buffer) {
+        if (Module['printChars'] != undefined && length) {
```

Comment:
Can do ` if (Module['printChars']` instead of ` if (Module['printChars'] != undefined`